aws/scenarios/vm/agentinstall/install.go

```diff
@@ -0,0 +1,74 @@
+package agentinstall
+
+import (
+	"fmt"
+
+	"github.com/DataDog/test-infra-definitions/aws"
+	"github.com/DataDog/test-infra-definitions/aws/scenarios/vm/os"
+	"github.com/DataDog/test-infra-definitions/command"
+	"github.com/DataDog/test-infra-definitions/common/utils"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func Install(runner *command.Runner, env aws.Environment, params *Params, os os.OS) error {
+	agentCommand := &agentCommand{version: params.version}
+	os.Visit(agentCommand)
+	cmd := agentCommand.cmd
+	lastCommand, err := runner.Command(
+		env.CommonNamer.ResourceName("agent-install", utils.StrHash(cmd)),
+		&command.CommandArgs{
+			Create: pulumi.Sprintf(cmd, params.apiKey),
+		})
+	if err != nil {
+		return err
+	}
+
+	agentConfig := ""
+	if params.agentConfig != "" {
+		fileManager := command.NewFileManager(runner)
+		agentConfig = fmt.Sprintf(params.agentConfig, params.apiKey)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
This means we expect `params.agentConfig` to be in the format 

```yaml
...
DD_API_KEY: %s
...
```

Have you considered receiving an agent config already formatted ? Or is there a way to be explicit about expecting agent config to be a template ?

Reply:
If this function accepts a whole Agent config, then the user has to manually adds the API key to this config which is not very convenient. 
Another option would be that `DD_API_KEY` is automatically inserting as a first line of the config. @pducolin what do you think?

Reply:
As a potential user, if I need to pass the agent config, I would probably pre-fill it correctly with the API key, so running the `Sprintf` from the caller side.

I find it counter intuitive to pass both the templated config and the api key, plus with `agentConfig` being required to respect a certain template. I would rather:

- pass only the api key -> a default template would be filled with the api key
- pass the whole config -> caller will take care of filling their template

that would translate in 

```golang
if params.agentConfig == "" && params.apiKey == "" {
  panic("one of agent config or apiKey must be valid")
}

agentConfig := params.agentConfig
if agentConfig == "" {
  // use default agent config template
  agentConfig = fmt.Sprintf(defaultAgentConfigTemplate, params.apiKey)
}
```

Does it make sense ?

Reply:
> pass only the api key -> a default template would be filled with the api key
You don't need to pass the API key if you use the default configuration so this case never happen.

I will update the code to pass the whole config.